### PR TITLE
chore(api): api image multistage build

### PR DESCRIPTION
**Pull request by JunIce**
Api image size reduce to 1.8 GB

### CODE REVIEW

File: api/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,15 @@
-FROM python:3.10-slim
+# packages install stage
+FROM python:3.10-slim AS base
 
 LABEL maintainer="takatost@gmail.com"
 
+COPY requirements.txt /requirements.txt
+
+RUN pip install --prefix=/pkg -r requirements.txt
```
Comment (takatost) [inline api/Dockerfile:11]: Before pip install, you need to install the following packages. These packages are required for the pip install process on the arm architecture. 
```
RUN apt-get update \
    && apt-get install -y --no-install-recommends gcc gcc g++ python3-dev libc-dev libffi-dev \
    && apt-get autoremove \
    && rm -rf /var/lib/apt/lists/*
```

File: api/Dockerfile [hunk 2]
```diff
@@ -15,13 +23,12 @@ EXPOSE 5001
...
 WORKDIR /app/api
 
-RUN apt-get update && \
-    apt-get install -y bash curl wget vim gcc g++ python3-dev libc-dev libffi-dev nodejs
-
-COPY requirements.txt /app/api/requirements.txt
-
-RUN pip install -r requirements.txt
+RUN apt-get update \ 
+    && apt-get install -y --no-install-recommends bash curl wget vim gcc g++ python3-dev libc-dev libffi-dev nodejs \
```
Comment (takatost) [inline api/Dockerfile]: This side can reduce the installation of the aforementioned dependencies, and it can be simplified to `bash curl wget vim nodejs`

**Review by takatost — CHANGES_REQUESTED** [review]

File: api/Dockerfile (hunk 2 above)
Comment (JunIce) [reply api/Dockerfile]: Have been modified.

**Comment by JunIce** [comment]
Why we need `vim` pkg, which scenario we use it?

**Comment by takatost** [comment]
> Why we need `vim` pkg, which scenario we use it?

You can remove it, it was originally for debugging purposes.

**Review by takatost — APPROVED** [review]
thx😊

**Comment by JunIce** [comment]
> You can remove it, it was originally for debugging purposes.

No need. I just want to know the details. Thank for your explaination.

**Pull request merged**